src/common/composables/amcharts5/index.ts

```diff
@@ -0,0 +1,56 @@
+import type { ComputedRef } from 'vue';
+import {
+    computed, reactive, toRefs, watchEffect,
+} from 'vue';
+
+import * as am5 from '@amcharts/amcharts5';
+import type { Root } from '@amcharts/amcharts5';
+import am5themes_Animated from '@amcharts/amcharts5/themes/Animated';
+
+import { createDonutChart, createPieChart, createPieSeries } from '@/common/composables/amcharts5/pie-chart-helper';
+import type { ChartContext } from '@/common/composables/amcharts5/type';
+import { createXYDateChart, createXYLineSeries, createXYStackedColumnSeries } from '@/common/composables/amcharts5/xy-chart-helper';
+
+export const useAmcharts5 = (
+    chartContext: ComputedRef<ChartContext>,
+) => {
+    const state = reactive({
+        root: computed(() => (chartContext.value ? am5.Root.new(chartContext.value) : undefined)),
+    });
+
+    const initRoot = (root: Root): void => {
+        root.setThemes([am5themes_Animated.new(root)]);
+        root.utc = true;
+    };
+
+    const disposeRoot = () => {
+        if (!state.root) return;
+        am5.array.each(am5.registry.rootElements, (d) => {
+            if (d === state.root) {
+                d.dispose();
+            }
+        });
+    };
+
+    const clearChildrenOfRoot = () => {
+        if (state.root) state.root.container.children.clear();
+    };
+
+    watchEffect(() => {
+        if (state.root) initRoot(state.root);
+    });
+
+    return {
+        ...toRefs(state),
+        disposeRoot,
+        clearChildrenOfRoot,
+        //
+        createXYDateChart,
+        createPieChart,
+        createDonutChart,
+        //
+        createXYLineSeries,
+        createXYStackedColumnSeries,
+        createPieSeries,
```

Comment:
이것들을 헬퍼함수로써 pure 함수이네요.
단독으로 쓰일 때 pure 함수의 이점은 분명히 존재하지만, 훅으로써는 그 의미를 잃는 것 같습니다.

그러지 않으려면 아래처럼 state.root를 주입하여 제공하는 것이 좋지 않을까 의견드려봅니다!

```suggestion
        createXYDateChart:  (settings) => { 
          if (!state.root) throw new Error('No Root')
          createXYDateChart(state.root, settings) 
        },
        ...
```

Reply:
오 넵!!!! 이쪽은 고민이 많았는데...다 pure 함수면 정말 훅을 만드는 의미가 없겠네요ㅎㅎ
바꾸겠습니다!